app/jobs/regular/notify_moved_posts.rb

```diff
@@ -6,27 +6,28 @@ def execute(args)
       raise Discourse::InvalidParameters.new(:post_ids) if args[:post_ids].blank?
       raise Discourse::InvalidParameters.new(:moved_by_id) if args[:moved_by_id].blank?
 
-      # Make sure we don't notify the same user twice (in case multiple posts were moved at once.)
-      users_notified = Set.new
       posts =
         Post
+          .includes(:user, :topic)
           .where(id: args[:post_ids])
           .where("user_id <> ?", args[:moved_by_id])
-          .includes(:user, :topic)
-      if posts.present?
-        moved_by = User.find_by(id: args[:moved_by_id])
+          .order(post_number: :asc)
+      return if posts.blank?
```

Comment:
I think `blank?` here (and `present?` before) causes an extra db query.

i.e. it does `SELECT 1 FROM …` here, and only later when `posts.each` is called it does the full query.

You'd have to either add `.to_a` after `.order()` or replace `blank?` with e.g. `none?`, I think?

(I'm about 75% certain about all the statements in this comment 😄)

Reply:
oh wow I did not know this!

Reply:
EDIT looked into it, seems like this isn't the case for a loaded query.